teamBangazon/BangazonCLI
# Addprodtoorder

Merge after pf-order/dylan7/kw-orderstringbuilder

### Changes Made
AddCustomerStringBuilderShould/AddPaymentTypeStringBuilderShould/AddProductStringbuilderShould:
Changed Method and Property names to fit restructure of files.

MainMenu:
Removed duplicate instantiation

DatabaseInterface;
Removed not null parameter from paymenttypeid to make it nullable

AddOrderManagerShould:
Created tests for AddProductToOrder and GetCurrentOrder
*in testing should assert.Equal that methods create expected orderProduct and order entries
ex.     AddProductToOrder(1, 1)
          Assert.Equal({expectedorderProductId}, result)
          if creating the first entry in orderProduct, entry should be an id of 1 with customer and product ids 
          of 1
ex.     GetCurrentOrder(1)
          Asser.Equal({expectedorderId}, result)
          if creating a new order should return newly created id, if one already open for customer, should 
          return that id

AddOrderManager:
Created 

### Steps to Test
```
git fetch --all
git checkout addprodtoorder
dotnet restore
```

*delete database if order table without nullable paymentId exists

`dotnet run`

add appropriate customers and products to be tested
run tests with ids of desired customers and products to test methods
ensure appropriate table rows are being created.

### Ref URL
[#5](https://github.com/teamBangazon/BangazonCLI/issues/5)
